pkg/wireguard/agent/agent.go

```diff
@@ -366,9 +340,21 @@ func (a *Agent) UpdatePeer(nodeName, pubKeyHex string, nodeIPv4, nodeIPv6 net.IP
 		var lookupIPv4, lookupIPv6 net.IP
 		if option.Config.EnableIPv4 && nodeIPv4 != nil {
 			lookupIPv4 = nodeIPv4
+			if a.nodeToNodeEncryption {
+				allowedIPs = append(allowedIPs, net.IPNet{
+					IP:   nodeIPv4,
+					Mask: net.CIDRMask(net.IPv4len*8, net.IPv4len*8),
```

Comment:
Maybe this is only unintuitive to me but I think `32` would be clearer than `net.IPv4len*8`.

Reply:
Cilium uses this [pattern elsewhere too](https://github.com/cilium/cilium/blob/48ddc9a494c756a7f6ba03b93bd1715dbd113f99/pkg/ip/ip.go#L18-L19), so I'd say let's keep it for now. We'll eventaully want to switch over to `netip.Addr` anyway, so it will likely be rewritten anyways. But I'd prefer to do that in a follow-up PR, since this one is already large enough.